### PR TITLE
Add support to container cluster resource for service external ips

### PR DESCRIPTION
**Pull request by googlerjk**
Added support to container cluster resource for service external ips

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: Added `service_external_ips_config` support to `cluster_container` resource.
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -3688,6 +3739,18 @@ func expandVerticalPodAutoscaling(configured interface{}) *container.VerticalPod
 	}
 }
 
+func expandServiceExternalIpsConfig(configured interface{}) *container.ServiceExternalIpsConfig {
```
Comment (c2thorn) [inline mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:3742]:
```suggestion
func expandServiceExternalIpsConfig(configured interface{}) *container.ServiceExternalIPsConfig {
```

File: mmv1/third_party/terraform/resources/resource_container_cluster.go.erb [hunk 2]
```diff
@@ -3688,6 +3739,18 @@ func expandVerticalPodAutoscaling(configured interface{}) *container.VerticalPod
...
+	l := configured.([]interface{})
+	if len(l) == 0 {
+		return nil
+	}
+	config := l[0].(map[string]interface{})
+	return &container.ServiceExternalIpsConfig{
```
Comment (c2thorn) [inline mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:3748]:
```suggestion
	return &container.ServiceExternalIPsConfig{
```

File: mmv1/third_party/terraform/resources/resource_container_cluster.go.erb [hunk 3]
```diff
@@ -4390,6 +4453,17 @@ func flattenResourceUsageExportConfig(c *container.ResourceUsageExportConfig) []
 	}
 }
 
+func flattenServiceExternalIpsConfig(c *container.ServiceExternalIpsConfig) []map[string]interface{} {
```
Comment (c2thorn) [inline mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:4456]:
```suggestion
func flattenServiceExternalIpsConfig(c *container.ServiceExternalIPsConfig) []map[string]interface{} {
```

**Review by c2thorn — CHANGES_REQUESTED** [review]
I think I figured out the issue. The variable name is lowercase P, but the type "*ServiceExternalIPsConfig" is upper case. I _think_ my suggested changes will fix

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 4]
```diff
@@ -2502,6 +2502,45 @@ func TestAccContainerCluster_errorNoClusterCreated(t *testing.T) {
...
+		CheckDestroy: testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_withExternalIpsConfigEnabled(pid, clusterName),
+			},
+			{
+				ResourceName:            "google_container_cluster.with_external_ips_config",
+				ImportState:             true,
+				ImportStateVerify:       true,
+			},
```
Comment (c2thorn) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:2523]: This step seems redundant? you already have `testAccContainerCluster_withExternalIpsConfig` with a value of `true`

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 5]
```diff
@@ -5236,6 +5275,42 @@ resource "google_container_cluster" "with_resource_labels" {
...
+	return fmt.Sprintf(`
+	data "google_project" "project" {
+		project_id = "%s"
+	}
+
+	resource "google_container_cluster" "with_external_ips_config" {
+		name               = "%s"
+		location           = "us-central1-a"
+		initial_node_count = 1
+		remove_default_node_pool = true
```
Comment (c2thorn) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:5289]: This line is causing your test to fail. it is a virtual field (isn't actually returned by the API). In the import step, the test runner expected this field to be set and returned by the API but it wasn't. Normally you'd have to set the field to be ignored, but if you just remove this whole function (see my comment above), that should also fix

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/website/docs/d/container_cluster.html.markdown [hunk 6]
```diff
@@ -2,59 +2,1092 @@
 subcategory: "Kubernetes (Container) Engine"
 page_title: "Google: google_container_cluster"
 description: |-
-  Get info about a Google Kubernetes Engine cluster.
```
Comment (c2thorn) [inline mmv1/third_party/terraform/website/docs/d/container_cluster.html.markdown:5]: Just caught that you are overwriting the datasource markdown file instead of the resource markdown file. Please undo the changes to this file and add your field documentation to the one under `docs/r/`

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb (hunk 5 above)
Comment (googlerjk) [reply mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb]: Fixed

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb (hunk 4 above)
Comment (googlerjk) [reply mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb]: Update applied

File: mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown [hunk 7]
```diff
@@ -427,6 +430,10 @@ addons_config {
   and `PROJECT_SINGLETON_POLICY_ENFORCE`. `PROJECT_SINGLETON_POLICY_ENFORCE` is functionally equivalent to the
   deprecated `enable_binary_authorization` parameter being set to `true`.
 
+<a name="nested_service_external_ips_config"></a>The `service_external_ips_config` block supports:
+
+* `service_external_ips_config` - (Required) Controls whether external ips specified by a service will be allowed. It is enabled by default.
```
Comment (c2thorn) [inline mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown:435]:
```suggestion
* `enabled` - (Required) Controls whether external ips specified by a service will be allowed. It is enabled by default.
```
The `service_external_ips_config` block supports `enabled` right?

**Review by c2thorn — CHANGES_REQUESTED** [review]

**Comment by googlerjk** [comment]
yes

**Review by c2thorn — APPROVED** [review]
